reactor-pool/src/test/java/reactor/pool/SimpleDequePoolInstrumentationTest.java

```diff
@@ -374,4 +377,66 @@ void testIsInactiveForMoreThan() {
 				.isTrue();
 	}
 
+	@Test
+	void testIssue172_block() throws InterruptedException {
+		Mono<Integer> allocator = Mono.just(1).subscribeOn(Schedulers.single());
+
+		InstrumentedPool<Integer> pool = PoolBuilder.from(allocator).sizeBetween(3, 7).buildPool();
+		pool.acquire().block();
+		pool.acquire().block();
+		assertThat(pool.metrics().allocatedSize()).isEqualTo(3);
+	}
+
+	@Test
+	void testIssue172_async() throws InterruptedException {
+		Mono<Integer> allocator = Mono.just(1).subscribeOn(Schedulers.single());
+
+		InstrumentedPool<Integer> pool = PoolBuilder.from(allocator).sizeBetween(3, 7).buildPool();
+		CountDownLatch latch = new CountDownLatch(2);
+		pool.acquire().subscribe(integerPooledRef -> latch.countDown());
+		pool.acquire().subscribe(integerPooledRef -> latch.countDown());
+		latch.await(5, TimeUnit.SECONDS);
+		assertThat(pool.metrics().allocatedSize()).isEqualTo(3);
+	}
+
+	@Test
+	void testIssue172_async_concurrent_warmup() throws InterruptedException {
+		Mono<Integer> allocator = Mono.just(1).subscribeOn(Schedulers.single());
+
+		InstrumentedPool<Integer> pool = PoolBuilder.from(allocator).sizeBetween(3, 7, 3).buildPool();
+		CountDownLatch latch = new CountDownLatch(2);
+		pool.acquire().subscribe(integerPooledRef -> latch.countDown());
+		pool.acquire().subscribe(integerPooledRef -> latch.countDown());
+		latch.await(5, TimeUnit.SECONDS);
+		assertThat(pool.metrics().allocatedSize()).isEqualTo(3);
+	}
+
+	@Test
+	void testIssue172_async_with_delay_and_more_than_min() throws InterruptedException {
+		Mono<Integer> allocator = Mono.just(1).delayElement(Duration.ofSeconds(1)).subscribeOn(Schedulers.single());
```

Comment:
AFAIU the `subscribeOn` operator is unnecessary, as the `delayElement` operator introduces asynchrony when delivering the result. The concurrency of that by default is the number of processors available.

Reply:
indeed, subscribeOn is useless here, thanks.

I have removed it, please check 
https://github.com/reactor/reactor-pool/pull/173/commits/079c6357882226b98684f0e99e204cd108d7a86a